iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/utils/CommandDetectorBuilder.java

```diff
@@ -30,103 +30,134 @@
 
 public class CommandDetectorBuilder {
 
-  private List<CommandPredicate> predicates = new ArrayList<>();
+  private final List<CommandPredicate> predicates = new ArrayList<>();
+  private final List<Predicate<ArgumentResolution>> containsPredicates = new ArrayList<>();
+  private boolean isContainsCalled;
 
   public CommandDetectorBuilder with(Predicate<String> predicate) {
+    validateContainsIsNotCalled();
     addSingularPredicate(predicate, CommandPredicate.Type.MATCH);
     return this;
   }
 
   public CommandDetectorBuilder with(Collection<String> firstOf) {
+    validateContainsIsNotCalled();
     return with(firstOf::contains);
   }
 
   public CommandDetectorBuilder with(String expectedString) {
+    validateContainsIsNotCalled();
     return with(expectedString::equals);
   }
```

Comment:
Technically we don't need to call this method again here as it will be done in the subsequent call.
```suggestion
  public CommandDetectorBuilder with(Collection<String> firstOf) {
    return with(firstOf::contains);
  }

  public CommandDetectorBuilder with(String expectedString) {
    return with(expectedString::equals);
  }
```